gateway/ln-gateway/src/webserver.rs

```diff
@@ -97,6 +97,14 @@ async fn pay_invoice(
     Json(contract_id): Json<ContractId>,
 ) -> Result<impl IntoResponse, LnGatewayError> {
     debug!(%contract_id, "Received request to pay invoice");
-    rpc.send(contract_id).await?;
+
+    // TODO: Require clients to pass in a federation id on request to pay invoice
+    let federation_id = FederationId("".into());
```

Comment:
I think before we merge this we need to nail down what a "federation id" is. Using the name here isn't a good idea because presumably that could change.

Reply:
Yeap. Makes sense to further that discussion

Reply:
This was my attempt to plan for any change in the value we end up using as Federation ID. I defined the ID as a struct in gateway so changes to the value does not result in any churn in gateway code:
```
// Placeholder struct for identifying federations within a gateway
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct FederationId(pub String);
```
IMO the signature should remain rather stable regardless of what we adopt. I didn't want to be super blocked by the discussion and implementation of the ID in Federation

Reply:
This was my attempt to plan for any change in the value we end up using as Federation ID. I defined the ID as a struct in gateway so changes to the value does not result in any churn in gateway code:
```
// Placeholder struct for identifying federations within a gateway
#[derive(Debug, Serialize, Deserialize, Clone)]
pub struct FederationId(pub String);
```
IMO the signature should remain rather stable regardless of what we adopt. I didn't want to be super blocked by the discussion and implementation of the ID in Federation

Reply:
For the purposes of this PR the federation id can be user defined afaik? You could just give it any id on join.

While I do have a slight preference for making this id deterministically derived from the config and using this right away as `FederationId(Sha256)` instead of `FederationId(String)` I'd be ok with this for now if it unblocks you and accelerates resolution of #807 which is the highest priority on the GW rn imo.

Reply:
Now switching fully to the high pri auth work. It's not related / blocked by this PR at all. Still, I wanted to leave this PR in a merge-able state so I don't have too many hanging conversations